I don't see how my change could be impacting this:

```
2023-09-25T13:08:14.8854475Z [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 13.21 s -- in io.quarkus.it.opentelemetry.MariaDbOpenTelemetryJdbcInstrumentationTest
2023-09-25T13:08:14.8888690Z [INFO] Running io.quarkus.it.opentelemetry.Db2OpenTelemetryJdbcInstrumentationTest
2023-09-25T13:08:14.9019947Z [WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.001 s -- in io.quarkus.it.opentelemetry.Db2OpenTelemetryJdbcInstrumentationTest
2023-09-25T13:08:14.9020997Z [INFO] Running io.quarkus.it.opentelemetry.OracleOpenTelemetryJdbcInstrumentationTest
2023-09-25T13:08:15.0667092Z 2023-09-25 13:08:14,908 INFO  [io.quarkus] (main) opentelemetry-jdbc-instrumentation-it stopped in 0.028s
2023-09-25T13:08:17.8923746Z 2023-09-25 13:08:17,800 WARN  [io.qua.hib.orm.dep.HibernateOrmProcessor] (build-21) Could not find a suitable persistence unit for model classes:
2023-09-25T13:08:17.8924595Z 	- io.quarkus.hibernate.orm.panache.PanacheEntity
2023-09-25T13:08:17.8925149Z 	- io.quarkus.it.opentelemetry.model.Hit
2023-09-25T13:08:18.5934819Z 2023-09-25 13:08:18,559 INFO  [JakartaTransformer] (build-15) Jakarta EE compatibility enhancer for Quarkus: transforming com.ibm.db2.jcc.t2zos.ab
2023-09-25T13:08:18.5936175Z 2023-09-25 13:08:18,564 INFO  [JakartaTransformer] (build-32) Jakarta EE compatibility enhancer for Quarkus: transforming com.ibm.db2.jcc.t2zos.T2zosConfiguration
2023-09-25T13:08:18.5937233Z 2023-09-25 13:08:18,566 INFO  [JakartaTransformer] (build-18) Jakarta EE compatibility enhancer for Quarkus: transforming com.ibm.db2.jcc.t2zos.T2zosConnection
2023-09-25T13:08:20.1184222Z 2023-09-25 13:08:20,116 INFO  [io.qua.hib.orm.run.JPAConfig] (main) Hibernate ORM persistence unit 'db2' was deactivated through configuration properties
2023-09-25T13:08:20.1185269Z 2023-09-25 13:08:20,117 INFO  [io.qua.hib.orm.run.JPAConfig] (main) Hibernate ORM persistence unit 'mariadb' was deactivated through configuration properties
2023-09-25T13:08:20.2185558Z 2023-09-25 13:08:20,119 INFO  [io.qua.hib.orm.run.JPAConfig] (main) Hibernate ORM persistence unit 'postgresql' was deactivated through configuration properties
2023-09-25T13:08:20.7253570Z 2023-09-25 13:08:20,686 WARN  [io.agr.pool] (agroal-11) Datasource 'oracle': ORA-00604: Error occurred at recursive SQL level 1. Check subsequent errors.
2023-09-25T13:08:20.7254642Z ORA-01804: failure to initialize timezone information
2023-09-25T13:08:20.7255575Z 
2023-09-25T13:08:20.7257628Z 2023-09-25 13:08:20,688 WARN  [org.hib.eng.jdb.env.int.JdbcEnvironmentInitiator] (JPA Startup Thread) HHH000342: Could not obtain connection to query metadata: java.sql.SQLException: ORA-00604: Error occurred at recursive SQL level 1. Check subsequent errors.
2023-09-25T13:08:20.7258735Z ORA-01804: failure to initialize timezone information
2023-09-25T13:08:20.7259207Z 
2023-09-25T13:08:20.7260904Z 	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:702)
2023-09-25T13:08:20.7318137Z 	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:603)
2023-09-25T13:08:20.7318633Z 	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:598)
2023-09-25T13:08:20.7319129Z 	at oracle.jdbc.driver.T4CTTIfun.processError(T4CTTIfun.java:1710)
2023-09-25T13:08:20.7319677Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.processError(T4CTTIoauthenticate.java:819)
2023-09-25T13:08:20.7320199Z 	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:1041)
2023-09-25T13:08:20.7320638Z 	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:443)
2023-09-25T13:08:20.7321130Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTH(T4CTTIoauthenticate.java:462)
2023-09-25T13:08:20.7321723Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTHWithO5Logon(T4CTTIoauthenticate.java:1628)
2023-09-25T13:08:20.7322331Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTH(T4CTTIoauthenticate.java:1375)
2023-09-25T13:08:20.7322888Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTH(T4CTTIoauthenticate.java:1322)
2023-09-25T13:08:20.7323479Z 	at oracle.jdbc.driver.T4CConnection.authenticateWithPassword(T4CConnection.java:1516)
2023-09-25T13:08:20.7324095Z 	at oracle.jdbc.driver.T4CConnection.authenticateUserForLogon(T4CConnection.java:1459)
2023-09-25T13:08:20.7324642Z 	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:828)
2023-09-25T13:08:20.7325146Z 	at oracle.jdbc.driver.PhysicalConnection.connect(PhysicalConnection.java:1132)
2023-09-25T13:08:20.7325725Z 	at oracle.jdbc.driver.T4CDriverExtension.getConnection(T4CDriverExtension.java:104)
2023-09-25T13:08:20.7326262Z 	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:837)
2023-09-25T13:08:20.7326729Z 	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:664)
2023-09-25T13:08:20.7327252Z 	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
2023-09-25T13:08:20.7327809Z 	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
2023-09-25T13:08:20.7328343Z 	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
2023-09-25T13:08:20.7328829Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
2023-09-25T13:08:20.7329669Z 	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
2023-09-25T13:08:20.7330348Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
2023-09-25T13:08:20.7330944Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2023-09-25T13:08:20.7331415Z 	at java.base/java.lang.Thread.run(Thread.java:1623)
2023-09-25T13:08:20.7331627Z 
2023-09-25T13:08:20.9079985Z 13:08:20.906 Oracle Database:2023-09-25T13:08:20.669466+00:00
2023-09-25T13:08:20.9081509Z 13:08:20.907 Oracle Database:FREEPDB1(3):TABLE AUDSYS.AUD$UNIFIED: ADDED INTERVAL PARTITION SYS_P351 (3374) VALUES LESS THAN (TIMESTAMP' 2023-09-26 00:00:00')
2023-09-25T13:08:21.0906032Z 2023-09-25 13:08:21,051 WARN  [io.agr.pool] (agroal-11) Datasource 'oracle': ORA-00604: Error occurred at recursive SQL level 1. Check subsequent errors.
2023-09-25T13:08:21.0906992Z ORA-01804: failure to initialize timezone information
2023-09-25T13:08:21.0907423Z 
2023-09-25T13:08:21.0908895Z 2023-09-25 13:08:21,052 WARN  [org.hib.eng.jdb.spi.SqlExceptionHelper] (JPA Startup Thread) SQL Error: 604, SQLState: 60000
2023-09-25T13:08:21.0909913Z 2023-09-25 13:08:21,052 ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (JPA Startup Thread) ORA-00604: Error occurred at recursive SQL level 1. Check subsequent errors.
2023-09-25T13:08:21.0922780Z ORA-01804: failure to initialize timezone information
2023-09-25T13:08:21.0922989Z 
2023-09-25T13:08:21.1802718Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.223 s <<< FAILURE! -- in io.quarkus.it.opentelemetry.OracleOpenTelemetryJdbcInstrumentationTest
2023-09-25T13:08:21.1803975Z [ERROR] io.quarkus.it.opentelemetry.OracleOpenTelemetryJdbcInstrumentationTest.testOracleQueryTraced -- Time elapsed: 0.004 s <<< ERROR!
2023-09-25T13:08:21.1804744Z java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
2023-09-25T13:08:21.1805370Z 	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:639)
2023-09-25T13:08:21.1806171Z 	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:723)
2023-09-25T13:08:21.1806774Z 	at java.base/java.util.Optional.orElseGet(Optional.java:364)
2023-09-25T13:08:21.1807199Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2023-09-25T13:08:21.1807607Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
2023-09-25T13:08:21.1808009Z Caused by: java.lang.RuntimeException: Failed to start quarkus
2023-09-25T13:08:21.1808428Z 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
2023-09-25T13:08:21.1808874Z 	at io.quarkus.runtime.Application.start(Application.java:101)
2023-09-25T13:08:21.1809296Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
2023-09-25T13:08:21.1809880Z 	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:282)
2023-09-25T13:08:21.1810482Z 	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:251)
2023-09-25T13:08:21.1811123Z 	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:606)
2023-09-25T13:08:21.1811736Z 	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:656)
2023-09-25T13:08:21.1812140Z 	... 1 more
2023-09-25T13:08:21.1823763Z Caused by: java.lang.RuntimeException: jakarta.persistence.PersistenceException: [PersistenceUnit: oracle] Unable to build Hibernate SessionFactory
2023-09-25T13:08:21.1824450Z 	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:78)
2023-09-25T13:08:21.1825168Z 	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:101)
2023-09-25T13:08:21.1825950Z 	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits1868654632.deploy_0(Unknown Source)
2023-09-25T13:08:21.1826641Z 	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits1868654632.deploy(Unknown Source)
2023-09-25T13:08:21.1827067Z 	... 8 more
2023-09-25T13:08:21.1827832Z Caused by: jakarta.persistence.PersistenceException: [PersistenceUnit: oracle] Unable to build Hibernate SessionFactory
2023-09-25T13:08:21.1828759Z 	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:126)
2023-09-25T13:08:21.1829827Z 	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:86)
2023-09-25T13:08:21.1830913Z 	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:74)
2023-09-25T13:08:21.1831819Z 	at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
2023-09-25T13:08:21.1832454Z 	at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
2023-09-25T13:08:21.1833043Z 	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:156)
2023-09-25T13:08:21.1833582Z 	at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:64)
2023-09-25T13:08:21.1834004Z 	at java.base/java.lang.Thread.run(Thread.java:1623)
2023-09-25T13:08:21.1834901Z Caused by: org.hibernate.exception.GenericJDBCException: Unable to open JDBC Connection for DDL execution [ORA-00604: Error occurred at recursive SQL level 1. Check subsequent errors.
2023-09-25T13:08:21.1835774Z ORA-01804: failure to initialize timezone information
2023-09-25T13:08:21.1836050Z ] [n/a]
2023-09-25T13:08:21.1836588Z 	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:61)
2023-09-25T13:08:21.1837317Z 	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
2023-09-25T13:08:21.1837953Z 	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:94)
2023-09-25T13:08:21.1838919Z 	at org.hibernate.resource.transaction.backend.jta.internal.DdlTransactionIsolatorJtaImpl.getIsolatedConnection(DdlTransactionIsolatorJtaImpl.java:77)
2023-09-25T13:08:21.1840076Z 	at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.getIsolatedConnection(GenerationTargetToDatabase.java:60)
2023-09-25T13:08:21.1841044Z 	at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.jdbcStatement(GenerationTargetToDatabase.java:110)
2023-09-25T13:08:21.1841933Z 	at org.hibernate.tool.schema.internal.exec.GenerationTargetToDatabase.accept(GenerationTargetToDatabase.java:77)
2023-09-25T13:08:21.1842637Z 	at org.hibernate.tool.schema.internal.Helper.applySqlString(Helper.java:235)
2023-09-25T13:08:21.1843224Z 	at org.hibernate.tool.schema.internal.Helper.applySqlStrings(Helper.java:219)
2023-09-25T13:08:21.1843869Z 	at org.hibernate.tool.schema.internal.SchemaDropperImpl.dropTables(SchemaDropperImpl.java:362)
2023-09-25T13:08:21.1844679Z 	at org.hibernate.tool.schema.internal.SchemaDropperImpl.dropConstraintsTablesSequences(SchemaDropperImpl.java:253)
2023-09-25T13:08:21.1845524Z 	at org.hibernate.tool.schema.internal.SchemaDropperImpl.dropFromMetadata(SchemaDropperImpl.java:215)
2023-09-25T13:08:21.1846260Z 	at org.hibernate.tool.schema.internal.SchemaDropperImpl.performDrop(SchemaDropperImpl.java:185)
2023-09-25T13:08:21.1846946Z 	at org.hibernate.tool.schema.internal.SchemaDropperImpl.doDrop(SchemaDropperImpl.java:155)
2023-09-25T13:08:21.1847608Z 	at org.hibernate.tool.schema.internal.SchemaDropperImpl.doDrop(SchemaDropperImpl.java:115)
2023-09-25T13:08:21.1848427Z 	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:242)
2023-09-25T13:08:21.1849355Z 	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.lambda$process$5(SchemaManagementToolCoordinator.java:143)
2023-09-25T13:08:21.1849974Z 	at java.base/java.util.HashMap.forEach(HashMap.java:1429)
2023-09-25T13:08:21.1850597Z 	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:140)
2023-09-25T13:08:21.1851766Z 	at io.quarkus.hibernate.orm.runtime.observers.SessionFactoryObserverForSchemaExport.sessionFactoryCreated(SessionFactoryObserverForSchemaExport.java:21)
2023-09-25T13:08:21.1852979Z 	at org.hibernate.internal.SessionFactoryObserverChain.sessionFactoryCreated(SessionFactoryObserverChain.java:35)
2023-09-25T13:08:21.1853679Z 	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:293)
2023-09-25T13:08:21.1854433Z 	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:84)
2023-09-25T13:08:21.1855010Z 	... 6 more
2023-09-25T13:08:21.1855526Z Caused by: java.sql.SQLException: ORA-00604: Error occurred at recursive SQL level 1. Check subsequent errors.
2023-09-25T13:08:21.1856693Z ORA-01804: failure to initialize timezone information
2023-09-25T13:08:21.1856905Z 
2023-09-25T13:08:21.1857134Z 	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:702)
2023-09-25T13:08:21.1857620Z 	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:603)
2023-09-25T13:08:21.1858107Z 	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:598)
2023-09-25T13:08:21.1858561Z 	at oracle.jdbc.driver.T4CTTIfun.processError(T4CTTIfun.java:1710)
2023-09-25T13:08:21.1859101Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.processError(T4CTTIoauthenticate.java:819)
2023-09-25T13:08:21.1859774Z 	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:1041)
2023-09-25T13:08:21.1860194Z 	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:443)
2023-09-25T13:08:21.1860686Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTH(T4CTTIoauthenticate.java:462)
2023-09-25T13:08:21.1861286Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTHWithO5Logon(T4CTTIoauthenticate.java:1628)
2023-09-25T13:08:21.1861875Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTH(T4CTTIoauthenticate.java:1375)
2023-09-25T13:08:21.1862826Z 	at oracle.jdbc.driver.T4CTTIoauthenticate.doOAUTH(T4CTTIoauthenticate.java:1322)
2023-09-25T13:08:21.1863423Z 	at oracle.jdbc.driver.T4CConnection.authenticateWithPassword(T4CConnection.java:1516)
2023-09-25T13:08:21.1864040Z 	at oracle.jdbc.driver.T4CConnection.authenticateUserForLogon(T4CConnection.java:1459)
2023-09-25T13:08:21.1864573Z 	at oracle.jdbc.driver.T4CConnection.logon(T4CConnection.java:828)
2023-09-25T13:08:21.1865091Z 	at oracle.jdbc.driver.PhysicalConnection.connect(PhysicalConnection.java:1132)
2023-09-25T13:08:21.1865656Z 	at oracle.jdbc.driver.T4CDriverExtension.getConnection(T4CDriverExtension.java:104)
2023-09-25T13:08:21.1866179Z 	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:837)
2023-09-25T13:08:21.1866642Z 	at oracle.jdbc.driver.OracleDriver.connect(OracleDriver.java:664)
2023-09-25T13:08:21.1867160Z 	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
2023-09-25T13:08:21.1867713Z 	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:536)
2023-09-25T13:08:21.1868237Z 	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:517)
2023-09-25T13:08:21.1868733Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
2023-09-25T13:08:21.1869304Z 	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
2023-09-25T13:08:21.1869960Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
2023-09-25T13:08:21.1870559Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
2023-09-25T13:08:21.1870941Z 	... 1 more
2023-09-25T13:08:21.1871075Z 
2023-09-25T13:08:21.3391464Z [INFO] 
2023-09-25T13:08:21.3394346Z [INFO] Results:
2023-09-25T13:08:21.3394917Z [INFO] 
2023-09-25T13:08:21.3395292Z [ERROR] Errors: 
2023-09-25T13:08:21.3396411Z [ERROR]   OracleOpenTelemetryJdbcInstrumentationTest.testOracleQueryTraced » Runtime java.lang.RuntimeException: Failed to start quarkus
2023-09-25T13:08:21.3397189Z [INFO] 
2023-09-25T13:08:21.3397613Z [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 1
```

cc @yrodiere

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/CustomObjectMapperTest.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/CustomObjectMapperTest.java
@@ -1,11 +1,14 @@
 package io.quarkus.resteasy.reactive.jackson.deployment.test;
 
 import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
 import static org.hamcrest.CoreMatchers.equalTo;
 
 import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
 
 import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
 import jakarta.ws.rs.POST;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.core.MediaType;
@@ -18,6 +21,7 @@ import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
 
 import io.quarkus.arc.Unremovable;
 import io.quarkus.test.QuarkusUnitTest;
@@ -32,26 +36,99 @@ public class CustomObjectMapperTest {
      * `objectMapper.enable(SerializationFeature.WRAP_ROOT_VALUE);`
      */
     @Test
-    void serverShouldUnwrapRootElement() {
-        given().body("{\"Request\":{\"value\":\"good\"}}")
+    void test() {
+        given().body("{\"Request\":{\"value\":\"FIRST\"}}")
                 .contentType(ContentType.JSON)
-                .post("/server")
+                .post("/server/dummy")
                 .then()
                 .statusCode(HttpStatus.SC_OK)
-                .body(equalTo("good"));
+                .body(equalTo("0"));
+
+        // ContextResolver was invoked for both reader and writer
+        when().get("/server/count")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("2"));
+
+        given().body("{\"Request2\":{\"value\":\"FIRST\"}}")
+                .contentType(ContentType.JSON)
+                .post("/server/dummy2")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("0"));
+
+        // ContextResolver was invoked for both reader and writer because different types where used
+        when().get("/server/count")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("4"));
+
+        given().body("{\"Request\":{\"value\":\"FIRST\"}}")
+                .contentType(ContentType.JSON)
+                .post("/server/dummy")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("0"));
+
+        // ContextResolver was not invoked because the types have already been cached
+        when().get("/server/count")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("4"));
+
+        given().body("{\"Request2\":{\"value\":\"FIRST\"}}")
+                .contentType(ContentType.JSON)
+                .post("/server/dummy2")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("0"));
+
+        // ContextResolver was not invoked because the types have already been cached
+        when().get("/server/count")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("4"));
+    }
+
+    private static void doTest() {
+
     }
 
     @Path("/server")
     public static class MyResource {
         @POST
         @Consumes(MediaType.APPLICATION_JSON)
-        public String post(Request request) {
-            return request.value;
+        @Path("dummy")
+        public Dummy dummy(Request request) {
+            return Dummy.valueOf(request.value);
+        }
+
+        @POST
+        @Consumes(MediaType.APPLICATION_JSON)
+        @Path("dummy2")
+        public Dummy2 dummy2(Request2 request) {
+            return Dummy2.valueOf(request.value);
+        }
+
+        @GET
+        @Path("count")
+        public long count() {
+            return CustomObjectMapperContextResolver.COUNT.get();
         }
     }
 
+    public enum Dummy {
+        FIRST,
+        SECOND
+    }
+
+    public enum Dummy2 {
+        FIRST,
+        SECOND
+    }
+
     public static class Request {
-        private String value;
+        protected String value;
 
         public Request() {
 
@@ -85,14 +162,21 @@ public class CustomObjectMapperTest {
         }
     }
 
+    public static class Request2 extends Request {
+    }
+
     @Provider
     @Unremovable
     public static class CustomObjectMapperContextResolver implements ContextResolver<ObjectMapper> {
 
+        static final AtomicLong COUNT = new AtomicLong();
+
         @Override
         public ObjectMapper getContext(final Class<?> type) {
+            COUNT.incrementAndGet();
             final ObjectMapper objectMapper = new ObjectMapper();
-            objectMapper.enable(DeserializationFeature.UNWRAP_ROOT_VALUE);
+            objectMapper.enable(DeserializationFeature.UNWRAP_ROOT_VALUE)
+                    .enable(SerializationFeature.WRITE_ENUMS_USING_INDEX);
             return objectMapper;
         }
     }

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/FullyFeaturedServerJacksonMessageBodyReader.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/FullyFeaturedServerJacksonMessageBodyReader.java
@@ -43,7 +43,8 @@ public class FullyFeaturedServerJacksonMessageBodyReader extends JacksonBasicMes
     private final Providers providers;
     private final ConcurrentMap<String, ObjectReader> perMethodReader = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, ObjectReader> perTypeReader = new ConcurrentHashMap<>();
-    private final ConcurrentMap<ObjectMapper, ObjectReader> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Class<?>, ObjectMapper> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<ObjectMapper, ObjectReader> objectReaderMap = new ConcurrentHashMap<>();
 
     @Inject
     public FullyFeaturedServerJacksonMessageBodyReader(ObjectMapper mapper, Providers providers) {
@@ -154,7 +155,7 @@ public class FullyFeaturedServerJacksonMessageBodyReader extends JacksonBasicMes
         ObjectReader effectiveReader = defaultReader;
         if (effectiveMapper != originalMapper) {
             // Effective reader based on the context
-            effectiveReader = contextResolverMap.computeIfAbsent(effectiveMapper, new Function<>() {
+            effectiveReader = objectReaderMap.computeIfAbsent(effectiveMapper, new Function<>() {
                 @Override
                 public ObjectReader apply(ObjectMapper objectMapper) {
                     return objectMapper.reader();
@@ -201,7 +202,16 @@ public class FullyFeaturedServerJacksonMessageBodyReader extends JacksonBasicMes
             contextResolver = providers.getContextResolver(ObjectMapper.class, null);
         }
         if (contextResolver != null) {
-            return contextResolver.getContext(type);
+            var cr = contextResolver;
+            ObjectMapper result = contextResolverMap.computeIfAbsent(type, new Function<>() {
+                @Override
+                public ObjectMapper apply(Class<?> aClass) {
+                    return cr.getContext(type);
+                }
+            });
+            if (result != null) {
+                return result;
+            }
         }
 
         return originalMapper;

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/FullyFeaturedServerJacksonMessageBodyWriter.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/FullyFeaturedServerJacksonMessageBodyWriter.java
@@ -38,7 +38,8 @@ public class FullyFeaturedServerJacksonMessageBodyWriter extends ServerMessageBo
     private final ObjectWriter defaultWriter;
     private final ConcurrentMap<String, ObjectWriter> perMethodWriter = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, ObjectWriter> perTypeWriter = new ConcurrentHashMap<>();
-    private final ConcurrentMap<ObjectMapper, ObjectWriter> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Class<?>, ObjectMapper> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<ObjectMapper, ObjectWriter> objectWriterMap = new ConcurrentHashMap<>();
 
     @Inject
     public FullyFeaturedServerJacksonMessageBodyWriter(ObjectMapper mapper, Providers providers) {
@@ -112,7 +113,7 @@ public class FullyFeaturedServerJacksonMessageBodyWriter extends ServerMessageBo
         if (effectiveMapper == originalMapper) {
             return defaultWriter;
         }
-        return contextResolverMap.computeIfAbsent(effectiveMapper, new Function<>() {
+        return objectWriterMap.computeIfAbsent(effectiveMapper, new Function<>() {
             @Override
             public ObjectWriter apply(ObjectMapper objectMapper) {
                 return createDefaultWriter(effectiveMapper);
@@ -133,7 +134,13 @@ public class FullyFeaturedServerJacksonMessageBodyWriter extends ServerMessageBo
             contextResolver = providers.getContextResolver(ObjectMapper.class, null);
         }
         if (contextResolver != null) {
-            ObjectMapper mapperFromContextResolver = contextResolver.getContext(o.getClass());
+            var cr = contextResolver;
+            ObjectMapper mapperFromContextResolver = contextResolverMap.computeIfAbsent(o.getClass(), new Function<>() {
+                @Override
+                public ObjectMapper apply(Class<?> aClass) {
+                    return cr.getContext(o.getClass());
+                }
+            });
             if (mapperFromContextResolver != null) {
                 effectiveMapper = mapperFromContextResolver;
             }

--- a/extensions/resteasy-reactive/rest-client-reactive-jackson/deployment/src/test/java/io/quarkus/rest/client/reactive/jackson/test/DifferentObjectMapperForClientAndServerTest.java
+++ b/extensions/resteasy-reactive/rest-client-reactive-jackson/deployment/src/test/java/io/quarkus/rest/client/reactive/jackson/test/DifferentObjectMapperForClientAndServerTest.java
@@ -3,13 +3,11 @@ package io.quarkus.rest.client.reactive.jackson.test;
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.net.URI;
 import java.util.Objects;
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
 
 import jakarta.inject.Singleton;
 import jakarta.ws.rs.GET;
@@ -70,10 +68,17 @@ public class DifferentObjectMapperForClientAndServerTest {
      */
     @Test
     void shouldClientUseCustomObjectMapperUnwrappingRootElement() {
-        assertFalse(ClientObjectMapperUnwrappingRootElement.USED.get());
+        AtomicLong count = ClientObjectMapperUnwrappingRootElement.COUNT;
+        assertEquals(0, count.get());
         Request request = clientUnwrappingRootElement.get();
         assertEquals("good", request.value);
-        assertTrue(ClientObjectMapperUnwrappingRootElement.USED.get());
+        assertEquals(1, count.get());
+
+        assertEquals("good", clientUnwrappingRootElement.get().value);
+        assertEquals("good", clientUnwrappingRootElement.get().value);
+        assertEquals("good", clientUnwrappingRootElement.get().value);
+        // count should not change as the resolution of the ObjectMapper should be cached
+        assertEquals(1, count.get());
     }
 
     /**
@@ -82,10 +87,17 @@ public class DifferentObjectMapperForClientAndServerTest {
      */
     @Test
     void shouldClientUseCustomObjectMapperNotUnwrappingRootElement() {
-        assertFalse(MyClientNotUnwrappingRootElement.CUSTOM_OBJECT_MAPPER_USED.get());
+        AtomicLong count = MyClientNotUnwrappingRootElement.CUSTOM_OBJECT_MAPPER_COUNT;
+        assertEquals(0, count.get());
         Request request = clientNotUnwrappingRootElement.get();
         assertNull(request.value);
-        assertTrue(MyClientNotUnwrappingRootElement.CUSTOM_OBJECT_MAPPER_USED.get());
+        assertEquals(1, count.get());
+
+        assertNull(clientNotUnwrappingRootElement.get().value);
+        assertNull(clientNotUnwrappingRootElement.get().value);
+        assertNull(clientNotUnwrappingRootElement.get().value);
+        // count should not change as the resolution of the ObjectMapper should be cached
+        assertEquals(1, count.get());
     }
 
     @Path("/server")
@@ -108,14 +120,14 @@ public class DifferentObjectMapperForClientAndServerTest {
     @Path("/server")
     @Produces(MediaType.APPLICATION_JSON)
     public interface MyClientNotUnwrappingRootElement {
-        AtomicBoolean CUSTOM_OBJECT_MAPPER_USED = new AtomicBoolean(false);
+        AtomicLong CUSTOM_OBJECT_MAPPER_COUNT = new AtomicLong();
 
         @GET
         Request get();
 
         @ClientObjectMapper
         static ObjectMapper objectMapper(ObjectMapper defaultObjectMapper) {
-            CUSTOM_OBJECT_MAPPER_USED.set(true);
+            CUSTOM_OBJECT_MAPPER_COUNT.incrementAndGet();
             return defaultObjectMapper.copy()
                     .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
                     .disable(DeserializationFeature.UNWRAP_ROOT_VALUE);
@@ -158,11 +170,11 @@ public class DifferentObjectMapperForClientAndServerTest {
     }
 
     public static class ClientObjectMapperUnwrappingRootElement implements ContextResolver<ObjectMapper> {
-        static final AtomicBoolean USED = new AtomicBoolean(false);
+        static final AtomicLong COUNT = new AtomicLong();
 
         @Override
         public ObjectMapper getContext(Class<?> type) {
-            USED.set(true);
+            COUNT.incrementAndGet();
             return new ObjectMapper().enable(DeserializationFeature.UNWRAP_ROOT_VALUE);
         }
     }

--- a/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyReader.java
+++ b/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyReader.java
@@ -1,5 +1,7 @@
 package io.quarkus.rest.client.reactive.jackson.runtime.serialisers;
 
+import static io.quarkus.rest.client.reactive.jackson.runtime.serialisers.JacksonUtil.getObjectMapperFromContext;
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.lang.annotation.Annotation;
@@ -13,8 +15,6 @@ import jakarta.ws.rs.WebApplicationException;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.MultivaluedMap;
 import jakarta.ws.rs.core.Response;
-import jakarta.ws.rs.ext.ContextResolver;
-import jakarta.ws.rs.ext.Providers;
 
 import org.jboss.logging.Logger;
 import org.jboss.resteasy.reactive.ClientWebApplicationException;
@@ -33,7 +33,8 @@ public class ClientJacksonMessageBodyReader extends JacksonBasicMessageBodyReade
 
     private static final Logger log = Logger.getLogger(ClientJacksonMessageBodyReader.class);
 
-    private final ConcurrentMap<ObjectMapper, ObjectReader> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<ResolverMapKey, ObjectMapper> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<ObjectMapper, ObjectReader> objectReaderMap = new ConcurrentHashMap<>();
     private RestClientRequestContext context;
 
     @Inject
@@ -66,43 +67,16 @@ public class ClientJacksonMessageBodyReader extends JacksonBasicMessageBodyReade
     }
 
     private ObjectReader getEffectiveReader(Class<Object> type, MediaType responseMediaType) {
-        ObjectMapper effectiveMapper = getObjectMapperFromContext(type, responseMediaType);
+        ObjectMapper effectiveMapper = getObjectMapperFromContext(type, responseMediaType, context, contextResolverMap);
         if (effectiveMapper == null) {
             return getEffectiveReader();
         }
 
-        return contextResolverMap.computeIfAbsent(effectiveMapper, new Function<>() {
+        return objectReaderMap.computeIfAbsent(effectiveMapper, new Function<>() {
             @Override
             public ObjectReader apply(ObjectMapper objectMapper) {
                 return objectMapper.reader();
             }
         });
-    }
-
-    private ObjectMapper getObjectMapperFromContext(Class<Object> type, MediaType responseMediaType) {
-        Providers providers = getProviders();
-        if (providers == null) {
-            return null;
-        }
-
-        ContextResolver<ObjectMapper> contextResolver = providers.getContextResolver(ObjectMapper.class,
-                responseMediaType);
-        if (contextResolver == null) {
-            // TODO: not sure if this is correct, but Jackson does this as well...
-            contextResolver = providers.getContextResolver(ObjectMapper.class, null);
-        }
-        if (contextResolver != null) {
-            return contextResolver.getContext(type);
-        }
-
-        return null;
-    }
-
-    private Providers getProviders() {
-        if (context != null && context.getClientRequestContext() != null) {
-            return context.getClientRequestContext().getProviders();
-        }
-
-        return null;
     }
 }

--- a/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java
+++ b/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ClientJacksonMessageBodyWriter.java
@@ -1,5 +1,6 @@
 package io.quarkus.rest.client.reactive.jackson.runtime.serialisers;
 
+import static io.quarkus.rest.client.reactive.jackson.runtime.serialisers.JacksonUtil.getObjectMapperFromContext;
 import static org.jboss.resteasy.reactive.server.jackson.JacksonMessageBodyWriterUtil.createDefaultWriter;
 import static org.jboss.resteasy.reactive.server.jackson.JacksonMessageBodyWriterUtil.doLegacyWrite;
 
@@ -27,7 +28,8 @@ public class ClientJacksonMessageBodyWriter implements MessageBodyWriter<Object>
 
     protected final ObjectMapper originalMapper;
     protected final ObjectWriter defaultWriter;
-    private final ConcurrentMap<ObjectMapper, ObjectWriter> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<ResolverMapKey, ObjectMapper> contextResolverMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<ObjectMapper, ObjectWriter> objectWriterMap = new ConcurrentHashMap<>();
     private RestClientRequestContext context;
 
     @Inject
@@ -44,7 +46,7 @@ public class ClientJacksonMessageBodyWriter implements MessageBodyWriter<Object>
     @Override
     public void writeTo(Object o, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType,
             MultivaluedMap<String, Object> httpHeaders, OutputStream entityStream) throws IOException, WebApplicationException {
-        doLegacyWrite(o, annotations, httpHeaders, entityStream, getEffectiveWriter());
+        doLegacyWrite(o, annotations, httpHeaders, entityStream, getEffectiveWriter(type, mediaType));
     }
 
     @Override
@@ -52,22 +54,18 @@ public class ClientJacksonMessageBodyWriter implements MessageBodyWriter<Object>
         this.context = requestContext;
     }
 
-    protected ObjectWriter getEffectiveWriter() {
-        if (context == null) {
-            // no context injected when writer is not running within a rest client context
-            return defaultWriter;
-        }
-
-        ObjectMapper objectMapper = context.getConfiguration().getFromContext(ObjectMapper.class);
+    protected ObjectWriter getEffectiveWriter(Class<?> type, MediaType responseMediaType) {
+        ObjectMapper objectMapper = getObjectMapperFromContext(type, responseMediaType, context, contextResolverMap);
         if (objectMapper == null) {
             return defaultWriter;
         }
 
-        return contextResolverMap.computeIfAbsent(objectMapper, new Function<>() {
+        return objectWriterMap.computeIfAbsent(objectMapper, new Function<>() {
             @Override
             public ObjectWriter apply(ObjectMapper objectMapper) {
                 return createDefaultWriter(objectMapper);
             }
         });
     }
+
 }

--- a/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/JacksonUtil.java
+++ b/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/JacksonUtil.java
@@ -1,0 +1,53 @@
+package io.quarkus.rest.client.reactive.jackson.runtime.serialisers;
+
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Function;
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.ext.ContextResolver;
+import jakarta.ws.rs.ext.Providers;
+
+import org.jboss.resteasy.reactive.client.impl.RestClientRequestContext;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+final class JacksonUtil {
+
+    private JacksonUtil() {
+    }
+
+    static ObjectMapper getObjectMapperFromContext(Class<?> type, MediaType responseMediaType, RestClientRequestContext context,
+            ConcurrentMap<ResolverMapKey, ObjectMapper> contextResolverMap) {
+        Providers providers = getProviders(context);
+        if (providers == null) {
+            return null;
+        }
+
+        ContextResolver<ObjectMapper> contextResolver = providers.getContextResolver(ObjectMapper.class,
+                responseMediaType);
+        if (contextResolver == null) {
+            // TODO: not sure if this is correct, but Jackson does this as well...
+            contextResolver = providers.getContextResolver(ObjectMapper.class, null);
+        }
+        if (contextResolver != null) {
+            var cr = contextResolver;
+            var key = new ResolverMapKey(type, context.getConfiguration(), context.getInvokedMethod().getDeclaringClass());
+            return contextResolverMap.computeIfAbsent(key, new Function<>() {
+                @Override
+                public ObjectMapper apply(ResolverMapKey resolverMapKey) {
+                    return cr.getContext(resolverMapKey.getType());
+                }
+            });
+        }
+
+        return null;
+    }
+
+    private static Providers getProviders(RestClientRequestContext context) {
+        if (context != null && context.getClientRequestContext() != null) {
+            return context.getClientRequestContext().getProviders();
+        }
+
+        return null;
+    }
+}

--- a/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ResolverMapKey.java
+++ b/extensions/resteasy-reactive/rest-client-reactive-jackson/runtime/src/main/java/io/quarkus/rest/client/reactive/jackson/runtime/serialisers/ResolverMapKey.java
@@ -1,0 +1,53 @@
+package io.quarkus.rest.client.reactive.jackson.runtime.serialisers;
+
+import java.util.Objects;
+
+import jakarta.ws.rs.core.Configuration;
+
+/**
+ * Each REST Client can potentially have different providers, so we need to make sure that
+ * caching for one client does not affect caching of another
+ */
+public final class ResolverMapKey {
+
+    private final Class<?> type;
+    private final Configuration configuration;
+
+    private final Class<?> restClientClass;
+
+    public ResolverMapKey(Class<?> type, Configuration configuration, Class<?> restClientClass) {
+        this.type = type;
+        this.configuration = configuration;
+        this.restClientClass = restClientClass;
+    }
+
+    public Class<?> getType() {
+        return type;
+    }
+
+    public Configuration getConfiguration() {
+        return configuration;
+    }
+
+    public Class<?> getRestClientClass() {
+        return restClientClass;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof ResolverMapKey)) {
+            return false;
+        }
+        ResolverMapKey that = (ResolverMapKey) o;
+        return Objects.equals(type, that.type) && Objects.equals(configuration, that.configuration)
+                && Objects.equals(restClientClass, that.restClientClass);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(type, configuration, restClientClass);
+    }
+}

